paasta_tools/paasta_metastatus.py

```diff
@@ -98,18 +98,21 @@ def get_extra_mesos_attribute_data(mesos_state):
     attributes = set().union(*(slave['attributes'].keys() for slave in mesos_state['slaves']))
 
     for attribute in attributes:
-        resource_dict = defaultdict(Counter)
+        resource_free_dict = defaultdict(Counter)
+        resource_availability_dict = dict()
         slave_attribute_mapping = {}
         for slave in mesos_state['slaves']:
             slave_attribute_name = slave['attributes'].get(attribute, 'UNDEFINED')
             slave_attribute_mapping[slave['id']] = slave_attribute_name
-            resource_dict[slave_attribute_name].update(filter_mesos_state_metrics(slave['resources']))
+            filtered_resources = filter_mesos_state_metrics(slave['resources'])
+            resource_free_dict[slave_attribute_name].update(filtered_resources)
+            resource_availability_dict[slave_attribute_name] = filtered_resources
```

Comment:
I don't think this will work -- you're not summing the total resources for each attribute type, just overwriting the `resource_availability_dict` for every host with that attribute.